Haskell/Blarney/Netlist/Passes/ConstantFold.hs

```diff
@@ -89,9 +89,9 @@ evalConstNet n@Net{..} = case n of
   -- Mux
   NetP (Mux _ w) (Lit s:xs) -> modNet (Identity w) [xs !! fromInteger s]
   -- Registers
-  NetP (RegisterEn i w) [Lit 0, _] -> modNet (Const w i) []
+  NetP (RegisterEn (Just i) w) [Lit 0, _] -> modNet (Const w i) []
```

Comment:
Can we also generate a `DontCare` node in case of `Nothing` init val and `Lit 0` enable?